build_rules.gradle

```diff
@@ -201,6 +201,13 @@ ext.applyJavaNature = {
     useJUnit { }
   }
 
+  // Ensure that tests are packaged and part of the artifact set.
```

Comment:
Is this required for release? Otherwise, the reason this isn't the default is because it is usually an error to depend on someone else's test jar. The only case in our project that is not a bug is the way we use NeedsRunner and ValidatesRunner.

Reply:
Cool - not that maven central listings are a terribly important UI but it doesn't really make sense to release them and it does clutter the namespace with misleading stuff, so I'll see about first eliminating the wonky deps and then no longer shipping them. No action on this PR, which is super savings.

Reply:
Its not about just eliminating the dependencies but shipping all the relevant testing utilities that may be currently contained within the test sourceset.

Reply:
Yes, shipping test utilities in a test jar is an error/bug to be fixed.